MBBSEmu/HostProcess/MbbsHost.cs

```diff
@@ -1184,14 +1206,36 @@ private void PatchRelocation(MbbsModule module)
             }
         }
 
-        private void ProcessNightlyCleanup()
+        private void EnableModule(string moduleId)
         {
-            if (_globalCache.ContainsKey("SYSOPGLOBAL-CLEANUP"))
+            var _moduleId = moduleId;
+            var moduleChange = _moduleConfigurations.FirstOrDefault(m => m.ModuleIdentifier.Equals(_moduleId, StringComparison.InvariantCultureIgnoreCase));
+
+            AddModule(new MbbsModule(_fileUtility, Clock, Logger, moduleChange.ModuleIdentifier, moduleChange.ModulePath) { MenuOptionKey = moduleChange.MenuOptionKey });
+
+            var moduleIndex = _moduleConfigurations.FindIndex(i => i.ModuleIdentifier.Equals(_moduleId, StringComparison.InvariantCultureIgnoreCase));
+            _moduleConfigurations[moduleIndex].ModuleEnabled = true;
+        }
+
+        private void DisableModule(string moduleId)
+        {
+            var _moduleId = moduleId;
+            var moduleChange = _modules.FirstOrDefault(m => m.Value.ModuleIdentifier.Equals(_moduleId, StringComparison.InvariantCultureIgnoreCase));
+
+            CallModuleRoutine("finrou", null, moduleChange.Value);
+            _modules.Remove(moduleChange.Value.ModuleIdentifier);
+            foreach (var e in _exportedFunctions.Keys.Where(x => x.StartsWith(moduleChange.Value.ModuleIdentifier)))
             {
-                _globalCache.Remove("SYSOPGLOBAL-CLEANUP");
-                _performCleanup = true;
+                _exportedFunctions[e].Dispose();
+                _exportedFunctions.Remove(e);
```

Comment:
modifying the collection while iterating is a good way to cause concurrency exceptions. Probably safer to store all items to be removed in a temp list and then remove all from the temp list